Mapsui.Rendering.Skia/SkiaWidgets/ScaleBarWidgetRenderer.cs

```
@@ -128,7 +128,7 @@ public class ScaleBarWidgetRenderer : ISkiaWidgetRenderer
                 canvas.DrawText(scaleBarText2, posX2, posY2 - textSize2.Top, _paintScaleTextStroke);
                 canvas.DrawText(scaleBarText2, posX2, posY2 - textSize2.Top, _paintScaleText);
 
-                envelop = envelop.Join(new MRect(posX2, posY2, posX2 + textSize2.Width, posY2 + textSize2.Height));
+                envelop = envelop?.Join(new MRect(posX2, posY2, posX2 + textSize2.Width, posY2 + textSize2.Height));
             }
 
             scaleBar.Envelope = envelop;
```

scaleBar.Envelope could now be null if the points are null. Better to throw an exception when there are no points. A scaleBar needs a position. And perhaps the nullable code can prevent null points altogether.